src/modules/0.0.21/aws_security_group/entity/index.ts

```diff
@@ -97,4 +106,14 @@ export class SecurityGroupRule {
     nullable: true,
   })
   description?: string;
+
+  @Check(
+    'Check_security_or_ip_permissions',
+    `("source_security_group" IS NULL AND ("from_port" IS NOT NULL OR "to_port" IS NOT NULL OR "cidr_ipv4" IS NOT NULL OR "cidr_ipv6" IS NOT NULL)) OR ("source_security_group" IS NOT NULL AND (("from_port" IS NULL OR "from_port"=-1) AND ("to_port" IS NULL OR "to_port"=-1) AND ("cidr_ipv4" IS NULL OR "cidr_ipv4"='0.0.0.0/0') AND ("cidr_ipv6" IS NULL)))`,
```

Comment:
two points here that confuse me:
- in the first part when `"source_security_group" IS NULL`. Is it possible to have only `from_port` defined and not the rest shouldn't they be `AND`s instead of `OR`s?
- in the second part, when checking the `cidr_ipv4` `NULL` or `0.0.0.0/0`  aren't they completely the opposite? is null or everything can pass?

Reply:
ok so first part, i think that i could refine a bit more, and set from_port IS NOT NULL AND to_port IS NOT NULL AND (cidr_ipv4 IS NOT NULL OR cidr_ipv6) IS NOT NULL. I will test that way
so 0.0.0.0/0 is the default value returned by AWS when there is a rule with a source security group. They are sending -1 on the ports, and 0.0.0.0/0 on cidr_ipv4